### ISSUE TITLE
[Enhancement] Provide hooks into hover/click for plugin extensibility.

### ISSUE THREAD

**Issue by exlted** [flot/flot#1602]
Currently, I am working on an internal version of the [valueLabel](https://github.com/winne27/flot-valuelabels) plugin which adds hover/click support to the labels.

We have our version "working", but is utilizing a likely unstable (for the long term) methodology to do so. It would be great if there was an officially supported method for extending interactivity in this manner. 

If there's already a supported/intended method for this, and I just glossed over it, I'd be grateful for the help.
And if a decision is made to push for implementing this, I'd be glad to work on the development of the change.

(Potentially partially related to #329)

Thanks!

**Comment by andrewdove1** [comment]
There is a plug-in jquery.flot.hover but not any hooks. Maybe you can provide an outline of how you would like proceed (or a PR containing prototype code) that could be reviewed.

**Comment by exlted** [comment]
My idea was to add a hook so plugins could register their version of findNearbyItem to be called during triggerClickHoverEvent. But this brings in the complication of potentially having multiple items being hovered over at once. 

To have minimal changes to the current hoverable/clickable spec the case with multiple would likely require adding another parameter which holds an array of all items that were provided by plugins, but I'm not sure how to choose which item would be sent as the "singular" item in that situation.

**Comment by andrewdove1** [comment]
I would try to use the same logic that is already in place for the 'singular' item ie. choose the topmost.

**Comment by exlted** [comment]
Not sure how to select the topmost, unless that means select the first one that's seen. I don't believe there's a way to detect "layering" unless you already know about all the items you're checking against. 

I have everything seemingly working besides that currently.

**Comment by andrewdove1** [comment]
You are correct - I meant the closest which is what is described in the documentation for findNearbyItem. If two are equally close then they would end up in the array in the same order as the hooks were called.

**Comment by exlted** [comment]
Makes sense, I have the changes pending in my fork. I'll do some more testing this evening, document, and if I think it's ready get a PR put together.
Probably need to finalize the name for the hook though, "clickHoverFindNearby" is, a little inelegant for my tastes.
